### PR TITLE
feat(args): improve api dx

### PR DESCRIPTION
**Pull request by CxRes**
As discussed in #482. Please accept this as well as some of the other very sensible pending PRs.

Breaking Change: The default platform is set as the current OS (using the prexisting OS detection mechanism). This prevents additional binaries from being downloaded when in run-mode apart from being a more sensible default.

Fixed: Arguments to be passed to NW.js instance must be specified after `--` flag (cli) or `options.argv` (js API). There were corner cases where this was not happening.

Modified: `--enable-logging` flag has been removed as a default argument to NW.js spawn. (User can pass this on their own volition or we can use a `--verbose` flag to do it for her. Even now, `--silent` is not linked to  `--enable-logging`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,29 +24,35 @@ Yes, there is also a [Grunt Plugin](https://github.com/nwjs/grunt-nw-builder). F
...
 
 ```js
 var NwBuilder = require('nw-builder');
 var nw = new NwBuilder({
     files: './path/to/nwfiles/**/**', // use the glob format
-    platforms: ['osx64', 'win32', 'win64'],
-    version: '0.14.6'
+    platforms: ['osx64', 'win32', 'win64', 'linux32', 'linux64'],
+    version: 'latest',
+    argv: '-- <args>' // see nwjs docs for possible <args>
```
Comment (zaygraveyard) [inline README.md]: Isn't this supposed to be `argv: [/*'<nwjs_arg>', ...*/]` (array vs string)

Comment (CxRes) [reply README.md]: Good Catch!!!!

File: README.md [hunk 2]
```diff
@@ -24,29 +24,35 @@ Yes, there is also a [Grunt Plugin](https://github.com/nwjs/grunt-nw-builder). F
...
 
 ```js
 var NwBuilder = require('nw-builder');
 var nw = new NwBuilder({
     files: './path/to/nwfiles/**/**', // use the glob format
-    platforms: ['osx64', 'win32', 'win64'],
-    version: '0.14.6'
+    platforms: ['osx64', 'win32', 'win64', 'linux32', 'linux64'],
+    version: 'latest',
+    argv: '[<nwjs_arg>, ... ]' // see nwjs docs for possible <nwjs_arg> values
```
Comment (zaygraveyard) [inline README.md]: Still a string, should be `argv: ['<nwjs_arg>', ...]` (the quotes are around the `<nwjs_arg>` not the entire array)

File: README.md [hunk 3]
```diff
@@ -88,7 +94,7 @@ The path to your node webkit app. It supports [simple-glob](https://github.com/j
 
 
 #### options.version
-Type: `String`  
+Type: `Array``String`
```
Comment (zaygraveyard) [inline README.md]: Typo?

Comment (CxRes) [reply README.md]: Oh Crap!!!

File: README.md (hunk 2 above)
Comment (CxRes) [reply README.md]: There should be no quotes anywhere actually. `'<x>'` is a literally <x>.

**Comment by CxRes** [comment]
Why don't you ask @adam-lynch to make you a maintainer????

**Comment by zaygraveyard** [comment]
> Why don't you ask @adam-lynch to make you a maintainer????

With great power comes great responsibility.
But I can certainly help.
If @adam-lynch has very little time to review PRs and/or merge them, I'll accept this power and help out.

**Comment by CxRes** [comment]
ROFL! You are maintaining a lib not saving planet earth....

But @zaygraveyard you have an eye for details... Which is important!!!!!

**Comment by CxRes** [comment]
I think the PR is rather sensible for the reasons I explained in the first post in this thread as well as #482. @zaygraveyard has also reviewed this and broadly agrees afaik.

**Comment by kessler** [comment]
@CxRes Thanks for your quick response, I agree with all the points raised in #482 and I want to merge your PR. Is it possible for your to fix the conflicts? I know it's asking much since a lot of time has gone by, so it's perfectly understandable if you'll say no.

**Comment by CxRes** [comment]
@kessler You'll have to give me some time. As it was so long ago, I do not even remember now the changes I had made. In particular, I am not sure about the last of three conflicts. I'll get back to you once I get a chance to have a look again!

**Comment by kessler** [comment]
@CxRes I don't have to give you anything that is inherently yours! :-) I'll wait patiently.

**Comment by CxRes** [comment]
@kessler Done... You can pull now!!!

**Pull request merged**